### PR TITLE
[Doc] Note for HeteroGraphConv / Doc build problems

### PR DESCRIPTION
**Pull request by NiklasBeierl**
## Description
This started out with #3003. I actually just wanted to add a note to the HeteroGraphConv module but stumbled into a bunch of problems with building the docs which I would like to fix / dcomument.

### The list of packages required for building the docs is incomplete.
- I extended the list in the `docs/README.md`, but I still couldn't get the docs successfully built (see below), so there might be more requirements luring.
- This list of requirements for building the docs is currently documented in two places `docs/source/contribute.rst` and `docs/README.md`. I think it would be better to document it in one place and have the other refer to it. I'd opt to keep it in `contributing.rst` and only keep additional details and a reference in `docs/README.md`.  Objections?
- Additionally I would suggest putting the requirements for the docs into a `requirements.txt` file. Objections?  

### Building the docs fails when following the steps from the contribution guide:
<details>
<summary> I get this error when building the docs: </summary>
Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/home/niklas/code/dgl/.venv/lib/python3.8/site-packages/sphinx/config.py", line 327, in eval_config_file
    exec(code, namespace)
  File "/home/niklas/code/dgl/docs/source/conf.py", line 27, in <module>
    import dgl
  File "/home/niklas/code/dgl/python/dgl/__init__.py", line 13, in <module>
    from .backend import load_backend, backend_name
  File "/home/niklas/code/dgl/python/dgl/backend/__init__.py", line 95, in <module>
    load_backend(get_preferred_backend())
  File "/home/niklas/code/dgl/python/dgl/backend/__init__.py", line 41, in load_backend
    from .._ffi.base import load_tensor_adapter # imports DGL C library
  File "/home/niklas/code/dgl/python/dgl/_ffi/base.py", line 10, in <module>
    from . import libinfo
  File "/home/niklas/code/dgl/python/dgl/_ffi/libinfo.py", line 96, in <module>
    find_lib_path()
  File "/home/niklas/code/dgl/python/dgl/_ffi/libinfo.py", line 84, in find_lib_path
    raise RuntimeError(message)
RuntimeError: Cannot find the files.
List of candidates:
/home/niklas/code/dgl/python/dgl/libdgl.so
/home/niklas/code/dgl/build/libdgl.so
/home/niklas/code/dgl/build/Release/libdgl.so
/home/niklas/code/dgl/lib/libdgl.so
/home/niklas/code/libdgl.so

make: *** [Makefile:22: mxnet] Error 2
</details>
It seems like dgl gets imported from the local files, but that fails because I didn't compile it. 
Can we avoid needing to compile the lib in order to build the docs?
If not, that should at least be pointed out in the contribution guide!  
I tried commenting out the `import dgl` from conf.py and was then at least able to start the build and figure out that some more packages need to be installed. 
<details>
<summary>
But it seems like some later stage of the doc build then imports dgl causing the same failure. 
</summary>
/home/niklas/code/dgl/tutorials/models/3_generative_model/5_dgmg.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/niklas/code/dgl/tutorials/models/3_generative_model/5_dgmg.py", line 53, in <module>
    import dgl
  File "/home/niklas/code/dgl/python/dgl/__init__.py", line 13, in <module>
    from .backend import load_backend, backend_name
  File "/home/niklas/code/dgl/python/dgl/backend/__init__.py", line 95, in <module>
    load_backend(get_preferred_backend())
  File "/home/niklas/code/dgl/python/dgl/backend/__init__.py", line 41, in load_backend
    from .._ffi.base import load_tensor_adapter # imports DGL C library
  File "/home/niklas/code/dgl/python/dgl/_ffi/base.py", line 10, in <module>
    from . import libinfo
  File "/home/niklas/code/dgl/python/dgl/_ffi/libinfo.py", line 96, in <module>
    find_lib_path()
  File "/home/niklas/code/dgl/python/dgl/_ffi/libinfo.py", line 84, in find_lib_path
    raise RuntimeError(message)
RuntimeError: Cannot find the files.
List of candidates:
/home/niklas/code/dgl/python/dgl/libdgl.so
/home/niklas/code/dgl/build/libdgl.so
/home/niklas/code/dgl/build/Release/libdgl.so
/home/niklas/code/dgl/lib/libdgl.so
/home/niklas/code/libdgl.so

-------------------------------------------------------------------------------
make: *** [Makefile:30: pytorch] Error 2
</details>

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

## Changes / TODOS
I will need some help / feedback from you guys with these: :) 
- [ ] The `libinfo.py` problem.
- [ ] Doc building process should be (properly) documented in **one** place.
- [ ] Adding requirements.txt for the docs.

### CODE REVIEW

File: python/dgl/nn/tensorflow/hetero.py [hunk 1]
```diff
@@ -88,6 +88,11 @@ class HeteroGraphConv(layers.Layer):
     >>> print(y_dst.keys())
     dict_keys(['user', 'game'])
 
+    .. note::
+    HeteroGraphConv requires that there is a module for every ``'etype'`` in an input graph.
+    If you want to apply HeteroGraphConv to a subset of a graphs ``'etypes'``, you must create a new graph using for
```
Comment (mufeili) [inline python/dgl/nn/tensorflow/hetero.py:93]: "graphs" -> "graph's"

File: python/dgl/nn/tensorflow/hetero.py [hunk 2]
```diff
@@ -88,6 +88,11 @@ class HeteroGraphConv(layers.Layer):
...
+    example: :func:`~dgl.edge_type_subgraph()`.
```
Comment (mufeili) [inline python/dgl/nn/tensorflow/hetero.py:94]: Remove ":" after "example"

**Comment by mufeili** [comment]
1. I guess `docs/source/requirements.txt` should be removed.
2. I've also left some minor comments.

File: python/dgl/nn/tensorflow/hetero.py [hunk 3]
```diff
@@ -88,6 +88,11 @@ class HeteroGraphConv(layers.Layer):
     >>> print(y_dst.keys())
     dict_keys(['user', 'game'])
 
+    .. note::
```
Comment (mufeili) [inline python/dgl/nn/tensorflow/hetero.py:91]: Have you tried building the doc and see if it gets rendered properly with `.. note::`?

**Comment by jermainewang** [comment]
> It seems like dgl gets imported from the local files, but that fails because I didn't compile it. Can we avoid needing to compile the lib in order to build the docs? If not, that should at least be pointed out in the contribution guide! I tried commenting out the `import dgl` from conf.py and was then at least able to start the build and figure out that some more packages need to be installed.

I think doc build currently needs DGL compiled from source unfortunately. Sorry for causing so many troubles! @mufeili if you feel the doc changes are reasonable, maybe we merge this first? I agree with @NiklasBeierl  that we should add a note to the README about the prerequisite to build DGL first. Moving forward, I feel we should have a mechanism for contributors to preview the doc change in github PR without building everything locally. cc @VoVAllen.

**Comment by mufeili** [comment]
@jermainewang I've updated this PR and will merge it.

**Review by mufeili — APPROVED** [review]

**Pull request merged**